src/Raven.Server/Documents/PeriodicBackup/Azure/RavenAzureClient.cs

```diff
@@ -480,21 +483,174 @@ List<BlobProperties> GetResult()
+            var url = $"{GetBaseServerUrl()}/?comp=batch";
+
+            var requestMessage = new HttpRequestMessage(HttpMethods.Post, url)
+            {
+                Headers =
                 {
-                    return listBlobsResult
-                        .Descendants("Blob")
-                        .Select(x => new BlobProperties
+                    { xMsDate, now }, { "x-ms-version", AzureStorageVersion }
+                }
+            };
+
+            var batchContent = new MultipartContent("mixed", $"batch_{Guid.NewGuid()}");
+            requestMessage.Content = batchContent;
+
+            var blobsWithIds = new Dictionary<string, string>();
+            for (var i = 0; i < blobs.Count; i++)
+            {
+                using var ms = new MemoryStream();
+                using var writer = new StreamWriter(ms);
+                var clientRequestId = Guid.NewGuid().ToString();
+                blobsWithIds[clientRequestId] = blobs[i];
+
+                await writer.WriteLineAsync($"{HttpMethods.Delete} /{_containerName}/{blobs[i]} HTTP/1.1");
+                await writer.WriteLineAsync($"{xMsDate}: {now}");
+                await writer.WriteLineAsync($"{xMsClientRequestId}: {clientRequestId}");
+                await writer.WriteLineAsync($"{xMsReturnClientRequestId}: true");
+
+                using (var hash = new HMACSHA256(_accountKey))
+                {
+                    var uri = new Uri($"{GetBaseServerUrl()}/{_containerName}/{blobs[i]}", UriKind.Absolute);
+                    var hashStr = $"{HttpMethods.Delete}\n\n\n\n\n\n\n\n\n\n\n\n{xMsClientRequestId}:{clientRequestId}\n{xMsDate}:{now}\n{xMsReturnClientRequestId}:true\n/{_accountName}{uri.AbsolutePath}";
+                    await writer.WriteLineAsync($"Authorization: SharedKey {_accountName}:{Convert.ToBase64String(hash.ComputeHash(Encoding.UTF8.GetBytes(hashStr)))}");
+                }
+
+                await writer.WriteLineAsync("Content-Length: 0");
+                await writer.FlushAsync();
+
+                batchContent.Add(new ByteArrayContent(ms.ToArray())
+                {
+                    Headers = { { "Content-Type", "application/http" }, { "Content-Transfer-Encoding", "binary" }, { "Content-ID", $"{i}" } }
+                });
+            }
+
+            var client = GetClient();
+
+            if (batchContent.Headers.ContentLength.HasValue == false)
+            {
+                // we need the ContentLength to CalculateAuthorizationHeaderValue
+                // the ContentLength is calculated on the fly, it gets added to Headers only when we try to access it.
+                throw new ArgumentException($"{nameof(MultipartContent)} should have content length");
+            }
+            client.DefaultRequestHeaders.Authorization = CalculateAuthorizationHeaderValue(HttpMethods.Post, url, requestMessage.Headers, batchContent.Headers);
+
+            var response = client.SendAsync(requestMessage).Result;
```

Comment:
we should use `await` here? The method is `async`

Reply:
done